src/frontend/src/contexts/LanguageContext.tsx

```diff
@@ -45,11 +45,25 @@ export const languages: Record<string, string> = {
 export function LanguageContext({ children }: { children: JSX.Element }) {
   const [language] = useLocalState((state) => [state.language]);
 
+  // workaround to initially set a locale with empty messages to not end up with an error
+  const [loaded, setLoaded] = useState(false);
+  const isMounted = useRef(true);
+
   useEffect(() => {
-    activateLocale(language);
+    isMounted.current = true;
+
+    activateLocale(language).then(() => {
+      if (isMounted.current) {
+        setLoaded(true);
+      }
+    });
+
+    return () => {
+      isMounted.current = false;
+    };
   }, [language]);
 
-  return <I18nProvider i18n={i18n}>{children}</I18nProvider>;
+  return loaded && <I18nProvider i18n={i18n}>{children}</I18nProvider>;
```

Comment:
Question: What is the state the page is left in if an error occurs while activating/mounting? Is it just an empty page then?

Reply:
Good catch, currently nothing will be shown. I'll change that.